foundations/src/telemetry/memory_profiler.rs

```diff
@@ -212,50 +192,50 @@ fn collect_heap_profile() -> Result<String> {
     Ok(String::from_utf8(profile)?)
 }
 
-#[cfg(feature = "security")]
-fn sandbox_jemalloc_syscalls() -> Result<()> {
-    #[cfg(target_arch = "x86_64")]
-    allow_list! {
-        static ALLOWED_SYSCALLS = [
-            ..SERVICE_BASICS,
-            // PXY-41: Required to call Instant::now from parking-lot.
-            clock_gettime,
-            openat,
-            creat,
-            unlink
-        ]
-    }
-
-    #[cfg(target_arch = "aarch64")]
-    allow_list! {
-        static ALLOWED_SYSCALLS = [
-            ..SERVICE_BASICS,
-            clock_gettime,
-            openat,
-            unlinkat
-        ]
-    }
-
-    enable_syscall_sandboxing(ViolationAction::KillProcess, &ALLOWED_SYSCALLS)?;
-
-    Ok(())
-}
-
 #[cfg(test)]
 mod tests {
     use super::*;
-    use crate::telemetry::settings::MemoryProfilerSettings;
+    use crate::{
+        security::{
+            allow_list,
+            common_syscall_allow_lists::{ASYNC, SERVICE_BASICS},
+            enable_syscall_sandboxing, ViolationAction,
+        },
+        telemetry::settings::MemoryProfilerSettings,
```

Comment:
Nit: don't use nested imports. Instead, introduce a separate `use` for each non-leaf path, e.g.:
```rust
use crate::security::{allow_list, enable_syscall_sandboxing, ViolationAction};
use crate::security::common_syscall_allow_lists::{ASYNC, SERVICE_BASICS};
...
```

Reply:
Ah the joys of auto importing. Too bad rustfmt's [imports_granularity = "Module"](https://rust-lang.github.io/rustfmt/?version=v1.6.0&search=Module) config isn't stabilized yet